src/transformers/pipelines/visual_question_answering.py

```diff
@@ -0,0 +1,165 @@
+                    return pipeline_inputs
+                elif isinstance(pipeline_inputs, dict):
+                    pipeline_inputs = [pipeline_inputs]
+                for pipeline_input in pipeline_inputs:
+                    if not isinstance(pipeline_input, dict):
+                        raise ValueError("Input is expected to be type `List[Dict[str, Union(str, Image.Image)]]`")
+                    question = pipeline_input.get("question", None)
+                    image = pipeline_input.get("image", None)
+                    if not isinstance(question, str):
+                        raise ValueError(
+                            "`question` is a required key in each input dict and expected value type is `str`"
+                        )
+                    if not (isinstance(image, str) or isinstance(image, Image.Image)):
+                        raise ValueError(
+                            "`image` is a required key in each input dict and expected value type is `str` or"
+                            " `Image.Image`"
+                        )
+            else:
+                raise ValueError("Please use keyword arguments `question` and `image`")
+        elif "question" in kwargs and "image" in kwargs:
+            if isinstance(kwargs["question"], list) and (
+                isinstance(kwargs["image"], str) or isinstance(kwargs["image"], Image.Image)
+            ):
+                pipeline_inputs = [{"question": Q, "image": kwargs["image"]} for Q in kwargs["question"]]
+            elif isinstance(kwargs["question"], str) and isinstance(kwargs["image"], list):
+                pipeline_inputs = [{"question": kwargs["question"], "image": image} for image in kwargs["image"]]
+            elif isinstance(kwargs["question"], list) and isinstance(kwargs["image"], list):
+                if len(kwargs["question"]) != len(kwargs["image"]):
+                    raise ValueError("Questions and images don't have the same lengths")
+                pipeline_inputs = [{"question": Q, "image": C} for Q, C in zip(kwargs["question"], kwargs["image"])]
+            elif isinstance(kwargs["question"], str) and (
+                isinstance(kwargs["image"], str) or isinstance(kwargs["image"], Image.Image)
+            ):
+                pipeline_inputs = [{"question": kwargs["question"], "image": kwargs["image"]}]
+            else:
+                raise ValueError("Arguments can't be understood")
+        else:
+            raise ValueError(f"Unknown arguments {kwargs}")
+        return pipeline_inputs
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class VisualQuestionAnsweringPipeline(Pipeline):
+    """
+    Visual Question Answering pipeline using a `ModelForVisualQuestionAnswering`. This pipeline is currently only
+    available in PyTorch.
+
+    This tabular question answering pipeline can currently be loaded from [`pipeline`] using the following task
+    identifier: `"visual-question-answering"`.
```

Comment:
```suggestion
    This visual question answering pipeline can currently be loaded from [`pipeline`] using the following task
    identifier: `"visual-question-answering"`.
```

Reply:
Either "vqa" or "visual-question-answering" works for me. I was mostly following the example of "tabular-question-answering"

Reply:
added the alias